src/screens/currency-list/currency-list.js

```diff
@@ -0,0 +1,59 @@
+import React, { useContext } from 'react';
+import { StatusBar, FlatList, Text } from 'react-native';
+import { useSafeArea } from 'react-native-safe-area-context';
+
+import { RowItem, RowSeperator } from '../../components/row-item';
+
+import { ItemPropTypes, propTypes } from './prop-types';
+
+import CURRENCIES from '../../utils/currencies.json';
+import { ConversionContext } from '../../utils';
+
+import { COLORS } from '../../styles';
+import { StyledWrapper, StyledFooterComponent } from './styled';
+
+export const CurrencyList = ({ navigation, route: { params } }) => {
+  const { baseCurrency, quoteCurrency, setBaseCurrency, setQuoteCurrency } = useContext(
+    ConversionContext,
+  );
+  const insets = useSafeArea();
+  const { isBaseCurrency } = params;
+
+  const Item = ({ item }) => {
+    let selected = false;
+    if (isBaseCurrency && item === baseCurrency) {
+      selected = true;
+    } else if (!isBaseCurrency && item === quoteCurrency) {
+      selected = true;
+    }
+    const navigateBack = () => {
+      if (isBaseCurrency) {
+        setBaseCurrency(item);
+      } else {
+        setQuoteCurrency(item);
+      }
+      navigation.pop();
+    };
+    return (
+      <RowItem text={item} onPress={navigateBack} rightIcon={selected && <Text>&#10004;</Text>} />
+    );
+  };
+  Item.propTypes = ItemPropTypes;
```

Comment:
Since it is behaving as a separate component now in isolation its better to move it to a separate file and import it to use in this list